lib/parslet/atoms/entity.rb

```diff
@@ -22,9 +23,11 @@ def try(source, context, consume_all)
   end
   
   def parslet
-    @parslet ||= @block.call.tap { |p| 
-      raise_not_implemented unless p
-    }
+    return @parslet unless @parslet.nil?
+    @parslet = @block.call
+    raise_not_implemented if @parslet.nil?
+    @parslet.label = @label
```

Comment:
The `try` method calls `parslet` so the object it returns must have its label set otherwise the error reporter won't be able to access it. The reporter specs create a fake atom that have a label set so don't test that. I've added a test to entity_spec.rb that will fail if these lines are removed.